qemu/tests/virtio_port_login_during_data_transfer.py

```diff
@@ -0,0 +1,100 @@
+import time
+import logging
+import threading
+import random
+from collections import deque
+
+from virttest import funcatexit
+from virttest import qemu_virtio_port
+from virttest.utils_virtio_port import VirtioPortTest
+
+from qemu.tests.virtio_port_login import ConsoleLoginTest
+
+EXIT_EVENT = threading.Event()
+
+
+def __set_exit_event():
+    """
+    Sets global EXIT_EVENT
+    :note: Used in cleanup by funcatexit
+    """
+    logging.info("Executing __set_exit_event()")
+    EXIT_EVENT.set()
+
+
+def run(test, params, env):
+    """
+    KVM virtio_console test
+
+    1. boot guest with virtio-serial port and virtio-console
+    2. login the guest via virtio-console
+    3. transfer data via virtio-serial, during data transfer logout and re-login
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    virtio_test = VirtioPortTest(test, env, params)
+    vm, guest_worker = virtio_test.get_vm_with_worker()
+    console = virtio_test.get_virtio_ports(vm)[0][0]
+    serialports = virtio_test.get_virtio_ports(vm)[1]
+    send_port = serialports[0]
+    recv_port = serialports[1]
+    params["login_console"] = console
+    login_test = ConsoleLoginTest(test, env, params)
+    logging.info("Login guest via virtio console: %s" % console)
+    #Login guest via console
+    session = login_test.console_login(console)
+
+    # Transfer data via serial port
+    threads = []
+    queues = []
+    queues.append(deque())
+    blocklen = 1024
+    send_port.open()
+    recv_port.open()
+    global EXIT_EVENT
+    funcatexit.register(env, params.get('type'), __set_exit_event)
+    thread = qemu_virtio_port.ThSendCheck(send_port, EXIT_EVENT,
+                                          queues, blocklen)
+    thread.start()
+    threads.append(thread)
+    thread = qemu_virtio_port.ThRecvCheck(recv_port, queues,
+                                          EXIT_EVENT, blocklen)
+    thread.start()
+    threads.append(thread)
+
+    # Logout and re-login the console session
+    def _close_session():
+        """Close session with random sleep before and after that"""
+        time.sleep(random.randrange(1, 20))
+        if session:
+            session.close()
+        time.sleep(random.randrange(1, 20))
+
+    _close_session()
+    session = login_test.console_login(console)
+    _close_session()
```

Comment:
Just logout -> re-login --> log out,  Do we need any other action after re-login?   To check data transfer is normal after re-login.

Reply:
@yanan-fu The case is to test transfer the data during this three action. The data transfer does not finish until L#82. As described in the comment:
```
 1. boot guest with virtio-serial port and virtio-console
  2. login the guest via virtio-console
  3. transfer data via virtio-serial, during data transfer logout and re-login
```
`ThSendCheck` and `ThRecvCheck` will check all the data transferred, no matter before or after the re-login, all the data transferred will be checked.